packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,718 @@
+  - [Technical Challenge in React Native](#technical-challenge-in-react-native)
+- [API Design Challenges](#api-design-challenges)
+  - [Approach: Let Consumer Compose `Popover`](#approach-let-consumer-compose-popover)
+    - [**1. Not possible to collocate the tour flows.**](#1-not-possible-to-collocate-the-tour-flows)
+    - [**2. How will we have two different tour flows in the same page?**](#2-how-will-we-have-two-different-tour-flows-in-the-same-page)
+    - [3. Segmentation / Lack of control](#3-segmentation--lack-of-control)
+    - [4. Not possible to maintain a consistent API / Implementation between web \& native](#4-not-possible-to-maintain-a-consistent-api--implementation-between-web--native)
+    - [Conclusion](#conclusion)
+- [MoMs](#moms)
+- [References](#references)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?type=design&node-id=40540-559304&t=tmTrf3xJU6oj59fM-0) to all variants of the Popover component
+
+## Features
+
+- [x] **Tour:** A guided tour with multiple steps
+- [x] **Masking:** Mask the rest of the page except the highlighted element.
+
+## API
+
+### `Tour` API
+
+```jsx
+// Step will have similar props as the Popover component, With extra Tour related props.
+type Step = {
+  /**
+   * Unique identifier for the tour step
+   */
+  name: string;
+  /**
+   * Content of the Popover
+   */
+  content: ({ gotToNext, goToPrev, stopTour, activeStep, totalSteps }) => React.ReactNode;
+  /**
+   * Footer content
+   */
+  footer?: ({ gotToNext, goToPrev, stopTour, activeStep, totalSteps }) => React.ReactNode;
+  /**
+   * Popover title
+   */
+  title?: string;
+  /**
+   * Leading content placed before the title
+   *
+   * Can be any blade icon or asset.
+   */
+  titleLeading?: React.ReactNode;
+  /**
+   * Placement of Popover
+   * @default "top"
+   */
+  placement?: UseFloatingOptions['placement'];
+}
+
+type TourProps = {
+  /**
+   * Array of steps to be rendered
+   *
+   * The order of the steps will be the order in which they are rendered depending on the `activeStep` prop
+   */
+  steps: Step[];
+  /**
+   * Whether the tour is visible or not
+   */
+  isOpen: boolean;
+  /**
+   * Callback when the tour is opened or closed
+   */
+  onOpenChange: ({ isOpen: boolean }) => void;
+  /**
+   * Callback which fires when the tour has reached the last step.
+   */
+  onFinish: () => void;
+  /**
+   * Callback when the active step changes
+   */
+  onStepChange?: (step: number) => void;
+  /**
+   * Active step to be rendered
+   */
+  activeStep: number;
+}
+```
+
+### `TourStep` API
+
+TourStep is an enhancer component which is used to wrap the element that needs to be highlighted with a specific unique identifier.
+
+This component is needed because in react-native there is no `id` prop or `getElementById` platform API, but to keep the API consistent between web & native, we will also expose this component for web.
```

Comment:
Also because we don't expose id from our components

Reply:
Yes will add this point.